source/Octopus.Tentacle.Client/TentacleClient.cs

```diff
@@ -209,6 +211,136 @@ async Task<DataStream> DownloadFileAction(CancellationToken ct)
             }
         }
 
+
+        public async Task<ScriptStatusResponseV3Alpha> StartScript(
+            StartScriptCommandV3Alpha startScriptCommand,
+            ILog logger,
+            CancellationToken scriptExecutionCancellationToken)
+        {
+            //TODO: sast - think about metrics. This isn't right, so we have ignored it for now. But we need to ask what we want to report on in an async event based world.
+            var operationMetricsBuilder = ClientOperationMetricsBuilder.Start();
+
+            var scriptServiceVersion = await DetermineScriptServiceVersionToUse(operationMetricsBuilder, logger, scriptExecutionCancellationToken);
+
+            var factory = new ScriptServiceExecutorFactory(
+                scriptServiceV1,
+                scriptServiceV2,
+                scriptServiceV3Alpha,
+                rpcCallExecutor,
+                operationMetricsBuilder,
+                OnCancellationAbandonCompleteScriptAfter,
+                clientOptions,
+                logger);
+
+            var executor = factory.CreateExecutor(scriptServiceVersion);
+
+            var result = await executor.StartScript(startScriptCommand, scriptExecutionCancellationToken);
+            return result;
+        }
+
+        public async Task<ScriptStatusResponseV3Alpha> GetStatus(
+            GetStatusCommandV3Alpha getStatusCommand,
+            ILog logger,
+            CancellationToken scriptExecutionCancellationToken)
+        {
+            //TODO: sast - think about metrics. This isn't right, so we have ignored it for now. But we need to ask what we want to report on in an async event based world.
+            var operationMetricsBuilder = ClientOperationMetricsBuilder.Start();
+
+            var scriptServiceVersion = await DetermineScriptServiceVersionToUse(operationMetricsBuilder, logger, scriptExecutionCancellationToken);
+
+            var factory = new ScriptServiceExecutorFactory(
+                scriptServiceV1,
+                scriptServiceV2,
+                scriptServiceV3Alpha,
+                rpcCallExecutor,
+                operationMetricsBuilder,
+                OnCancellationAbandonCompleteScriptAfter,
+                clientOptions,
+                logger);
+
+            var executor = factory.CreateExecutor(scriptServiceVersion);
+
+            var result = await executor.GetStatus(getStatusCommand, scriptExecutionCancellationToken);
+
+            return result;
+        }
```

Comment:
is the caller of this method now responsible for delays between calls?

Reply:
Yes 🙂